sdk/farmer/src/lib.rs

```diff
@@ -616,23 +665,87 @@ impl Config {
-            sector_plotting_handler_ids.push(single_disk_farm.on_sector_plotted(Arc::new(
-                move |(plotted_sector, maybe_old_plotted_sector)| {
+            sector_plotting_handler_ids.push(single_disk_farm.on_sector_update(Arc::new({
+                let single_disk_farm_id = *single_disk_farm.id();
+                let farmer_metrics = farmer_metrics.clone();
+
+                move |(_plotted_sector, sector_update)| {
                     let _span_guard = span.enter();
-                    handler_on_sector_plotted(
-                        plotted_sector,
-                        maybe_old_plotted_sector,
+                    let farmer_metrics = farmer_metrics.clone();
+
+                    match sector_update {
+                        SectorUpdate::Plotting(SectorPlottingDetails::Starting { .. }) => {
+                            farmer_metrics.sector_plotting.inc();
+                        }
+                        SectorUpdate::Plotting(SectorPlottingDetails::Downloading) => {
+                            farmer_metrics.sector_downloading.inc();
+                        }
+                        SectorUpdate::Plotting(SectorPlottingDetails::Downloaded(time)) => {
+                            farmer_metrics
+                                .observe_sector_downloading_time(&single_disk_farm_id, time);
+                            farmer_metrics.sector_downloaded.inc();
+                        }
+                        SectorUpdate::Plotting(SectorPlottingDetails::Encoding) => {
+                            farmer_metrics.sector_encoding.inc();
+                        }
+                        SectorUpdate::Plotting(SectorPlottingDetails::Encoded(time)) => {
+                            farmer_metrics.observe_sector_encoding_time(&single_disk_farm_id, time);
+                            farmer_metrics.sector_encoded.inc();
+                        }
+                        SectorUpdate::Plotting(SectorPlottingDetails::Writing) => {
+                            farmer_metrics.sector_writing.inc();
+                        }
+                        SectorUpdate::Plotting(SectorPlottingDetails::Written(time)) => {
+                            farmer_metrics.observe_sector_writing_time(&single_disk_farm_id, time);
+                            farmer_metrics.sector_written.inc();
+                        }
+                        SectorUpdate::Plotting(SectorPlottingDetails::Finished {
+                            plotted_sector: _,
+                            old_plotted_sector: _,
+                            time,
+                        }) => {
+                            farmer_metrics.observe_sector_plotting_time(&single_disk_farm_id, time);
+                            farmer_metrics.sector_plotted.inc();
+                        }
+                        _ => {}
+                    };
+
+                    handler_on_sector_update(
```

Comment:
You forward the whole update, but only care about plotted sector there. I'd leave it be `handler_on_sector_plotted` as before and forward just plotting update there like before. When/if you need more it can be extended.